vispy/gloo/glir.py

```diff
@@ -1193,10 +1193,10 @@ def delete(self):
 
     def set_framebuffer(self, yes):
         if yes:
+            self.activate()
             if not self._validated:
                 self._validated = True
                 self._validate()
-            self.activate()
```

Comment:
Turns out we did have a check for the framebuffer status, we were just doing the check *before* actually activating the framebuffer, which seems to be the wrong order of operations. (Now on Intel if I have a bad attachment, I get an appropriate error as below.)